core/http_bench.js

```diff
@@ -72,36 +29,73 @@ function createResolvable() {
   return Object.assign(promise, methods);
 }
 
+const scratch32 = new Int32Array(4);
+const scratchBytes = new Uint8Array(
+  scratch32.buffer,
+  scratch32.byteOffset,
+  scratch32.byteLength
+);
+assert(scratchBytes.byteLength === 4 * 4);
+
+// Toggle what method we send with. false = legacy.
+// AFAICT This has no effect on performance.
+const sendWithShared = true;
+
+function send(promiseId, opId, arg, zeroCopy = null) {
+  scratch32[0] = promiseId;
+  scratch32[1] = opId;
+  scratch32[2] = arg;
+  scratch32[3] = -1;
+  if (sendWithShared) {
+    Deno._sharedQueue.push(scratchBytes);
+    libdeno.send(null, zeroCopy);
+  } else {
+    libdeno.send(scratchBytes, zeroCopy);
+  }
+}
+
 /** Returns Promise<number> */
-function sendAsync(opId, arg, zeroCopyData) {
+function sendAsync(opId, arg, zeroCopy = null) {
   const promiseId = nextPromiseId++;
   const p = createResolvable();
-  recordsReset();
-  recordsPush(promiseId, opId, arg, -1);
   promiseMap.set(promiseId, p);
-  libdeno.send(null, zeroCopyData);
+  send(promiseId, opId, arg, zeroCopy);
   return p;
 }
 
-/** Returns u32 number */
-function sendSync(opId, arg) {
-  recordsReset();
-  recordsPush(0, opId, arg, -1);
-  libdeno.send();
-  if (recordsSize() != 1) {
-    throw Error("Expected sharedSimple to have size 1");
-  }
-  let { result } = recordsShift();
-  return result;
+function recordFromBuf(buf) {
+  assert(buf.byteLength === 16);
+  const buf32 = new Int32Array(buf.buffer, buf.byteOffset, buf.byteLength);
```

Comment:
3rd argument should be unit length, not byteLength.
When optimizing further, it's worth a try to create a global static Int32Array rather than making and disposing one every time. In my experience the difference is not huge but noticeable.

Reply:
fixed